swarm/README.md

```diff
@@ -0,0 +1,55 @@
+## Swarm
+
+[https://swarm.ethereum.org](https://swarm.ethereum.org)
+
+Swarm is a distributed storage platform and content distribution service, a native base layer service of the ethereum web3 stack. The primary objective of Swarm is to provide a decentralized and redundant store for dapp code and data as well as block chain and state data. Swarm is also set out to provide various base layer services for web3, including node-to-node messaging, media streaming, decentralised database services and scalable state-channel infrastructure for decentralised service economies.
+
+[![Travis](https://travis-ci.org/ethereum/go-ethereum.svg?branch=master)](https://travis-ci.org/ethereum/go-ethereum)
+[![Gitter](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/ethereum/swarm?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
+
+
+## Building the source
+
+Building Swarm requires Go (version 1.10 or later).
+
+    go get -d github.com/ethereum/go-ethereum
+
+    go install github.com/ethereum/go-ethereum/cmd/swarm
+
+
+## Documentation
+
+Swarm documentation can be found at [https://swarm-guide.readthedocs.io](https://swarm-guide.readthedocs.io).
+
+
+## Contribution
+
+Thank you for considering to help out with the source code! We welcome contributions from
+anyone on the internet, and are grateful for even the smallest of fixes!
+
+If you'd like to contribute to Swarm, please fork, fix, commit and send a pull request
+for the maintainers to review and merge into the main code base. If you wish to submit more
+complex changes though, please check up with the core devs first on [our Swarm gitter channel](https://gitter.im/ethereum/swarm)
+to ensure those changes are in line with the general philosophy of the project and/or get some
+early feedback which can make both your efforts much lighter as well as our review and merge
+procedures quick and simple.
+
+Please make sure your contributions adhere to our coding guidelines:
+
+ * Code must adhere to the official Go [formatting](https://golang.org/doc/effective_go.html#formatting) guidelines (i.e. uses [gofmt](https://golang.org/cmd/gofmt/)).
```

Comment:
there could be a lot more info here see swarm onboarding doc, dev guide, contribution guidelines

Reply:
Sure, we could enhance it further. I just wanted to have the minimum necessary as a start.

1. Contribution guidelines are here.
2. Onboarding document - I don't think it belongs on the front page of the README. We should also update that document. I checked it today and it is a bit outdated.
3. Dev guide - I am not sure what you mean by that?

Reply:
how about linking to and consolidate the following docs? 

* https://github.com/ethersphere/go-ethereum/wiki/Git-Workflow
* https://github.com/ethersphere/swarm/wiki/swarm
* https://github.com/ethereum/go-ethereum/wiki/Developers'-Guide 
* https://github.com/ethereum/go-ethereum#contribution

Reply:
I think we can do this step by step, however:
https://github.com/ethersphere/go-ethereum/wiki/Git-Workflow - I don't think we need this anymore
https://github.com/ethersphere/swarm/wiki/swarm - the URL and the whole document has to be refactored, but I agree we need an Onboarding document.
https://github.com/ethereum/go-ethereum/wiki/Developers'-Guide - I have to think about this one. We probably need something like this for Swarm.
https://github.com/ethereum/go-ethereum#contribution - this is already part of our README.

Bottom line - I suggest we merge as is, and continue with fixing the Developers' Guide and the Onboarding document.